### PR TITLE
Changes on butterflies placement checks

### LINKED ISSUE #2987: Butterflies don't respect other nodes on_rightclick
**Observed behavior**
When I try to open a chest with a butterfly in hand the butterfly is always placed.

**Expected behavior**
The butterfly is placed only if I hold the sneak key. Just like other placeable nodes.

Edit:
Just noticed that the fireflies have the same issue.

### LINKED ISSUE #3102: Avoid using the global keyword `table` as a function argument name
In the file [functions.lua](https://github.com/minetest/minetest_game/blob/master/mods/default/functions.lua), many functions use `table` as an argument name. Since `table` is also a global variable, it ends up being shadowed or redefined locally. This is generally regarded as a poor programming practice as it can be confusing what `table` is referring to.

### PR DESCRIPTION
**Pull request by JoseDouglas26**
Fixes #2987 (by using `minetest.item_place`)
Fixes #3102 (by renaming the parameters)

### CODE REVIEW

**Comment by JoseDouglas26** [comment]
Sorry, I made a commit to the master of my fork to resolve another issue and it ended up in this PR too. Now this PR can fix #3102

File: mods/butterflies/init.lua [hunk 1]
```diff
@@ -47,13 +47,16 @@ for i in ipairs (butter_list) do
 			local player_name = placer and placer:get_player_name() or ""
 			local pos = pointed_thing.above
 
-			if not minetest.is_protected(pos, player_name) and
-					not minetest.is_protected(pointed_thing.under, player_name) and
-					minetest.get_node(pos).name == "air" then
-				minetest.set_node(pos, {name = "butterflies:butterfly_"..name})
+			if minetest.is_protected(pos, player_name) or minetest.is_protected(pointed_thing.under, player_name) then
```
Comment (SmallJoker) [inline mods/butterflies/init.lua:50]: Please keep the lines below 80 characters.
https://dev.minetest.net/Lua_code_style_guidelines#Lines.2C_spaces.2C_and_indentation

Same for the hidden butterfly code.

File: mods/butterflies/init.lua [hunk 2]
```diff
@@ -47,13 +47,16 @@ for i in ipairs (butter_list) do
...
+				return
+			end
+
+			local _, sucess = minetest.item_place(itemstack, placer, pointed_thing)
```
Comment (SmallJoker) [inline mods/butterflies/init.lua:54]: Correction: `success`
Same for the hidden butterfly code.

File: mods/default/functions.lua [hunk 3]
```diff
@@ -2,149 +2,149 @@
 -- Sounds
 --
 
-function default.node_sound_defaults(table)
-	table = table or {}
-	table.footstep = table.footstep or
+function default.node_sound_defaults(tbl)
```
Comment (SmallJoker) [inline mods/default/functions.lua:5]: I'd rather name this parameter `sounds`, but I'll leave the decision up to you.

File: mods/butterflies/init.lua (hunk 1 above)
Comment (SmallJoker) [inline mods/butterflies/init.lua:50]: This protection check might not be necessary at all. `core.item_place_node` (called by `minetest.item_place`) already performs protection checks.

Comment (JoseDouglas26) [reply mods/butterflies/init.lua]: I wasn't sure if the check was done so I kept

File: mods/butterflies/init.lua (hunk 2 above)
Comment (JoseDouglas26) [reply mods/butterflies/init.lua]: Poor english, sorry about that

**Review by SmallJoker — APPROVED** [review]
I found that butterflies now do replace grass. In master, the butterfly is put to `pointed_thing.above`. However, this can be avoided by using sneak + place.

Works for me.

**Comment by Emojigit** [comment]
I submitted another PR, #3125, which uses the engine's routine to handle butterfly and firefly placement and catching. Every possible interaction exception can be handled by having everything cared for by the engine.

**Comment by JoseDouglas26** [comment]
So can I close this?

**Comment by Emojigit** [comment]
> So can I close this (again)?
> 
> 

This PR has been approved by core developers, so I would suggest letting them decide.

**Comment by JoseDouglas26** [comment]
I will close this PR. #3125 will be better thant that one

**Comment by JoseDouglas26** [comment]
I will close this PR. #3125 will be better thant that one

**Pull request closed without merging**